File: C++/Data Structures/Sorting/bubbleSort.cpp
#include<iostream>
using namespace std;

void swap(int *m,int *n){
    int temp = *m;
    *m = *n;
    *n = temp;
}

void printArray(int arr[],int n){
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

void bubbleSort(int arr[],int n){
    for(int i=0;i<n;i++){
        for(int j=0;j<n-i-1;j++){
            if(arr[j] > arr[j+1]){
                swap(arr[j],arr[j+1]);
            }
        }
    }
}

int main(void){
    int arr[] = {20,5,78,9,2,3,56,19};
    int n = sizeof(arr)/sizeof(arr[0]);
    //cout<<n<<endl;
    cout<<"The unsorted array is : ";
    printArray(arr,n);
    bubbleSort(arr,n);
    cout<<"After sorting : ";
    printArray(arr,n);
    return 0;
}

File: Python/Data Structures/Sorting/bubbleSort.py
def bubbleSort(arr):
    n = len(arr) # Length of Array

    # Traverse the array
    for i in range(n):
        for j in range(0,n-i-1):
            if arr[j] > arr[j+1]:
                arr[j],arr[j+1] = arr[j+1],arr[j]

def printArray(arr):
    for i in range(len(arr)):
        print(arr[i],end=" ")
    print()

if __name__ == '__main__':
    arr = [64,34,25,12,22,11,90]
    bubbleSort(arr)
    printArray(arr)